src/dune_data_loading.rs

```diff
@@ -10,9 +10,10 @@ use std::fs::{read_dir, File};
 use std::io::Read;
 use std::path::Path;
 
-pub fn load_dune_data_into_memory<P: AsRef<Path>>(path: P) -> Result<DatabaseStruct> {
+pub fn load_dune_data_into_memory<P: Clone + AsRef<Path>>(path: P) -> Result<DatabaseStruct> {
     let mut memory_database: HashMap<H160, Vec<Data>> = HashMap::new();
     let mut date = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(0, 0), Utc);
+    std::fs::create_dir_all(path.clone())?;
```

Comment:
I assume this does nothing if everything already exists?

Reply:
Yes, I am very sure about it.
from the docs:
>Recursively create a directory and all of its parent components if they are missing.